### PR TITLE
Notice: Undefined index: shipping_carrier_code is returned instead of  Required parameter "shipping_carrier_code" is missin

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
### Description (*)
Issue: https://github.com/magento/graphql-ce/issues/400

#### Preconditions (*)
1. Registered Customer
2. Virtual Product
3. Developer mode

#### Steps to reproduce (*)
1. Generate Customer token
2. Create empty cart as Registered Customer from GraphQL Client
3. Add virtual product to cart from GraphQL Client but use

```
...
    cart {
      items {
        id
...
```
in output
4. Get `ADDRESS_ID` by hovering 'Change Billing Address' or 'Change Shipping Address'

1. Execute:

```
mutation {
  setShippingMethodsOnCart(
    input: {
      cart_id: "{{ CART_ID }}"
      shipping_methods: {
        cart_address_id: {{ ADDRESS_ID }}
        method_code: "flatrate"
        carrier_code: "flatrate"
      }
    }
  ) {
    cart {
      available_payment_methods {
        code
        title
      }
      shipping_addresses {
        city
        available_shipping_methods {
          carrier_title
          error_message
          method_title
          method_code
          price_excl_tax
          price_incl_tax
        }
        selected_shipping_method {
          amount
        }
      }
    }
  }
}
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php [hunk 1]
```diff
@@ -69,13 +69,13 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
 
         $shippingMethod = reset($shippingMethods); // This point can be extended for multishipping
 
-        if (!$shippingMethod['cart_address_id']) {
+        if (!array_key_exists('cart_address_id', $shippingMethod)) {
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php]: We can replace with `isset` and `empty`
Please check, why we don't have notices during test running

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php (hunk 1 above)
Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php]: used 'isset' instead array_key_exists

File: app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php [hunk 2]
```diff
@@ -69,13 +69,13 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
 
         $shippingMethod = reset($shippingMethods); // This point can be extended for multishipping
 
-        if (!$shippingMethod['cart_address_id']) {
+        if (!isset($shippingMethod['cart_address_id'])) {
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php]: Also, need to check that value is not empty

Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Resolver/SetShippingMethodsOnCart.php]: add a condition to check empty value. Merge conflict was resolved

**Comment by naydav** [comment]
https://github.com/magento/graphql-ce/pull/405

**Pull request merged**